src/Package/Impl/Utilities/ProjectUtilities.cs

```diff
@@ -42,5 +52,95 @@ public static class ProjectUtilities {
 
             return result;
         }
+
+        public static EnvDTE.Project GetActiveProject() {
+            DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
+            var projects = (Array)dte.ActiveSolutionProjects;
+            if (projects != null && projects.Length == 1) {
+                return (EnvDTE.Project)projects.GetValue(0);
+            }
+            return null;
+        }
+
+        public static void AddNewItem(string templateName, string name, string extension, string destinationPath) {
```

Comment:
Do we really have to do the whole unzip dance ourselves? I hoped VS would have provided a programmatic way to do the same thing that "Add New Item" does when user clicks OK...